include/ginkgo/core/base/range_accessors.hpp

```diff
@@ -205,6 +635,301 @@ class row_major {
 };
 
 
+/**
+ * Namespace for helper functions and structs for
+ * the block column major accessor.
+ */
+namespace detail_block_col_major {
```

Comment:
Why is this namespace itself `detail_block_col_major` instead of `namespace detail { namespace block_col_major {}} ` ?

Reply:
Yes, I guess one option would be to put the row-major helpers in `detail::row_major::` and block-col-major helpers in `detail::block_col_major::`. Would that be better?

Reply:
Yes, I think that would be better.